posix/posix.c

```diff
@@ -806,6 +806,10 @@ int _posix_dup2(process_info_t *p, int fildes, int fildes2)
 		return -EBADF;
 	}
 
+	if (fildes == fildes2) {
+		return EOK;
```

Comment:
> If fildes2 is already a valid open file descriptor, it shall be closed first, unless fildes is equal to fildes2 in which case dup2() shall return fildes2 without closing it.

According to POSIX, `fildes2` should be returned in this case. `dup2()` returns fd or an error and sets errno, so `EOK` would be interpreted as fd=0

Reply:
Yeah, I ve been testing on fd == 0 so i didnt catch it thanks!